src/main/java/com/epam/pages/popup/SubjectPopup.java

```diff
@@ -1,6 +1,127 @@
+    }
+
+    public void fillExistedSubjectName() {
+        String existedName = SharedTestData.getLastCreatedItemName();
+        logger.info("Fill existed subject name {}", existedName);
+        uiHelper.sendKeys(nameField, existedName);
+    }
+
+    public void clickOnTheTeachersDropDownList() {
+        uiHelper.clickOnWebElement(teacherSelectField);
+    }
+
+    public void fillTeacherName(String teacherName) {
+        SharedTestData.setLastInputtedTeacherName(teacherName);
+        logger.info("Fill teacher name {}", teacherName);
+        uiHelper.sendKeys(teacherSelectField, teacherName);
+    }
+
+    public void selectTeacherByName(String teacherName) {
+        Select select = new Select(teacherSelect);
+        logger.info("Select teacher by name {}", teacherName);
+        select.selectByVisibleText(teacherName);
+    }
+
+    public void selectTeacher() {
+        logger.info("Select teacher");
+        Select select = new Select(teacherSelect);
+        select.selectByIndex(0);
+    }
+
+    public void saveValueFromSubjectNameField() {
+        SharedTestData.setLastCreatedItemName(nameField.getDomProperty("value"));
+    }
+
+    public void saveValueOfSelectedTeacher() {
+        SharedTestData.setLastCreatedItemName(listOfSelectedTeachers.get(listOfSelectedTeachers.size() - 1).getText());
+    }
+
+    public boolean checkAllElementsArePresent() {
+        return uiHelper.checkElementsAreDisplayed(
+                title,
+                xButton,
+                saveButton,
+                nameField,
+                teacherSelect
+        );
+    }
+
+    public boolean checkAllFieldsAndDropDownListAreEmptyInSubjectsSectionCreatePopup() {
+        logger.info("Check all fields and drop-down list are empty in subject section create popup");
+        return uiHelper.checkElementsAreEmpty(nameField) && !uiHelper.areElementsSelected(teacherSelect);
+    }
+
+    public String getExistedSubjectNameErrorMessage() {
+        return existedSubjectNameErrorMessage.getText();
+    }
+
+    public boolean checkMatchedItemsAreAppearedBelowTheSearchLine() {
+        logger.info("Check matched items are appeared below the search line");
+        return teachers.stream()
+                .allMatch(
+                        teacherName -> teacherName.getText().toLowerCase()
+                                .contains(SharedTestData.getLastInputtedTeacherName().toLowerCase())
+                );
+    }
+
+    public boolean checkSelectedItemsAreShownWithTheXIcon() {
+        logger.info("Check selected items are shown with the 'x' icon");
+        return listOfSelectedTeachers.stream()
+                .allMatch(WebElement::isDisplayed)
+                && xButtonsOfSelectedTeachers.stream()
+                .allMatch(WebElement::isDisplayed);
+    }
+
+    public void clickOnXButtonOfSelectedTeacher() {
+        uiHelper.clickOnWebElement(xButtonsOfSelectedTeachers.get(xButtonsOfSelectedTeachers.size() - 1));
```

Comment:
I suggest adding another method that will click on xButton of selected teachers until all selected items will be cleared... But it is not a must, or you can overload that method and add parameter size and clear selected teachers until expected size of list is reached